homeassistant/components/subaru/config_flow.py

```diff
@@ -0,0 +1,168 @@
+"""Config flow for Subaru integration."""
+from datetime import datetime
+import logging
+
+from subarulink import (
+    Controller as SubaruAPI,
+    InvalidCredentials,
+    InvalidPIN,
+    SubaruException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import (
+    CONF_HARD_POLL_INTERVAL,
+    DEFAULT_HARD_POLL_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    MIN_HARD_POLL_INTERVAL,
+    MIN_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Subaru instances."""
+    return {entry.title for entry in hass.config_entries.async_entries(DOMAIN)}
+
+
+class SubaruConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Subaru."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        error = None
+
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): str,
+                        vol.Required(CONF_PASSWORD): str,
+                        vol.Required(CONF_PIN): str,
+                    }
+                ),
+            )
+
+        if user_input[CONF_USERNAME] in configured_instances(self.hass):
+            return self.async_abort(reason="already_configured")
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except InvalidCredentials:
+            error = {"base": "invalid_auth"}
+        except InvalidPIN:
+            error = {"base": "invalid_pin"}
+        except SubaruException as ex:
+            _LOGGER.error("Unable to communicate with Subaru API: %s", ex.message)
+            return self.async_abort(reason="cannot_connect")
+
+        if error:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_USERNAME, default=user_input.get(CONF_USERNAME)
+                        ): str,
+                        vol.Required(
+                            CONF_PASSWORD, default=user_input.get(CONF_PASSWORD)
+                        ): str,
+                        vol.Required(CONF_PIN, default=user_input.get(CONF_PIN)): str,
+                    }
+                ),
+                errors=error,
+            )
+
+        return self.async_create_entry(title=user_input[CONF_USERNAME], data=info)
```

Comment:
You can do this in an `else:` after the `try:`

Reply:
Change made.